app/src/main/java/com/asfoundation/wallet/promotions/voucher/EVoucherDetailsInteractor.kt

```diff
@@ -0,0 +1,21 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import java.util.*
+
+class EVoucherDetailsInteractor {
+
+  fun getDiamondModels(): List<DiamondsButtonModel> {
+    val diamondsButtonModels: MutableList<DiamondsButtonModel> = LinkedList()
+    diamondsButtonModels.add(DiamondsButtonModel(43, "Diamonds"))
+    diamondsButtonModels.add(DiamondsButtonModel(218, "Diamonds"))
+    diamondsButtonModels.add(DiamondsButtonModel(430, "Diamonds"))
+    diamondsButtonModels.add(DiamondsButtonModel(43, "Diamonds"))
+    diamondsButtonModels.add(DiamondsButtonModel(218, "Diamonds"))
+    diamondsButtonModels.add(DiamondsButtonModel(430, "Diamonds"))
+    return diamondsButtonModels
+  }
+
+  fun getTitle(): String {
```

Comment:
From checking e-vouchers confluence specification, the title doesn't seem like it will come from here. It seems like it will be passed from the previous screen (where it lists all the vouchers), so this title should probably be passed as a fragment argument instead of being here.